### PR TITLE
feat(metrics): improve backpressure metrics

### PR DESCRIPTION
**Pull request by fuyufjh**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

See background #12345

image

image

PS. I am considering to add an Actor/Fragment-level switch on Grafana dashboard. In most cases, fragment-level would be more useful. Let's do it later.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/dispatch.rs [hunk 1]
```diff
@@ -66,40 +67,71 @@ impl DispatchExecutorInner {
...
+                    let start_time = Instant::now();
                     dispatcher.dispatch_barrier(barrier.clone()).await?;
+                    self.metrics
+                        .actor_output_buffer_blocking_duration_ns
+                        .with_label_values(&[
+                            &self.actor_id_str,
+                            &self.fragment_id_str,
+                            dispatcher.dispatcher_id_str(),
+                        ])
+                        .inc_by(start_time.elapsed().as_nanos() as u64);
```
Comment (fuyufjh) [inline src/stream/src/executor/dispatch.rs:130]: This part is verbose, but I failed to refactor it with a function/lambda due to the borrow checker 😇

Comment (BugenZhao) [reply src/stream/src/executor/dispatch.rs:130]: Then what about a macro? 🥵

**Review by BugenZhao — COMMENTED** [review]
LGTM!

**Review by BugenZhao — APPROVED** [review]

File: src/stream/src/executor/dispatch.rs (hunk 1 above)
Comment (fuyufjh) [reply src/stream/src/executor/dispatch.rs:130]: A little bit overkilling 😄

**Review by yezizp2012 — APPROVED** [review]
LGTM!

**Review by hzxa21 — APPROVED** [review]
LGTM

**Pull request merged**